### PR TITLE
I2013 - stochastic template button

### PR DESCRIPTION
**Pull request by littledice**
And a bit of opportunistic css tidying up

### CODE REVIEW

File: app/src/main/report/style.scss [hunk 1]
```diff
@@ -16,12 +16,6 @@ $montagu-breadcrumb-color: $success;
 @import "./styles/sidebar";
 @import "./styles/report-table";
 
-.download-icon {
```
Comment (littledice) [inline app/src/main/report/style.scss]: moved to shared scss file

File: app/src/main/shared/components/FileDownloadLink.tsx [hunk 2]
```diff
@@ -23,7 +23,7 @@ export class FileDownloadButtonInner extends React.Component<OneTimeLinkProps, u
 
     render() {
         return <FileDownloadInner
-            className="button"
+            className={"button" + (this.props.className ? ` ${this.props.className}` : "")}
```
Comment (littledice) [inline app/src/main/shared/components/FileDownloadLink.tsx]: to allow extra classnames passed as props

File: app/src/main/shared/components/FileDownloadLink.tsx [hunk 3]
```diff
@@ -57,7 +57,7 @@ export class FileDownloadInner extends React.Component<OneTimeLinkProps, undefin
             className = this.props.className;
             loader = null;
         } else {
-            className = this.props.className + ' disabledLink';
+            className = this.props.className + ' disabled';
```
Comment (littledice) [inline app/src/main/shared/components/FileDownloadLink.tsx]: removed custom class in favour of standard bootstrap one

File: app/src/test/shared/components/OneTimeLinkContextTests.tsx [hunk 4]
```diff
@@ -38,25 +38,31 @@ describe("OneTimeLinkContext", () => {
...
         const rendered = shallow(<Class href="/orange/"/>, {context: {store}}).dive();
         const child = rendered.find(EmptyComponent);
-        expect(child.prop("href")).to.equal(null);
+        checkAsync(done, () => {
+            expect(child.prop("href")).to.equal(null);
+        });
     });
 
-    it("can get properties from store with matching token", () => {
+    it("can get properties from store with matching token", (done: DoneCallback) => {
```
Comment (littledice) [inline app/src/test/shared/components/OneTimeLinkContextTests.tsx]: we were getting weird errors here due to asynchronous method being fired and not resolved before the test finished

**Comment by MartinEden** [comment]
I think you're right

**Pull request merged**